src/webgpu/shader/validation/variable_and_const.spec.ts

```diff
@@ -137,12 +137,37 @@ g.test('v_0038')
     const { storageClass, containerType, scalarType } = t.params;
     const type = containerType ? `${containerType}<${scalarType}>` : scalarType;
 
-    const code = `
-      [[location(0)]] var<${storageClass}> a : ${type} = ${type}();
+    var code;
+    if (`${storageClass}` === 'in') {
+      code = `
+        struct MyInputs {
+          [[location(0)]] a : ${type};
+        };
+
+        [[stage(vertex)]]
+        fn main(inputs : MyInputs) {
+        }
+      `;
+    } else if (`${storageClass}` === 'out') {
+      code = `
+        struct MyOutputs {
+          [[location(0)]] a : ${type};
```

Comment:
As above, this struct needs to include a position builtin.

Reply:
Actually, there's probably no reason this test needs to use a vertex shader, so could just make these all fragment shaders to keep things simpler.

Reply:
Converted to fragment